### PR TITLE
 presence: Validate presence aggreagated before considering it.

### PR DESCRIPTION
**Pull request by jainkuniya**
Let me know changes :)

### CODE REVIEW

File: src/users/userHelpers.js [hunk 1]
```diff
@@ -36,7 +37,14 @@ export const groupUsersByInitials = (users: User[]): Object =>
 export const groupUsersByStatus = (users: User[], presences: Object): Object =>
   users.reduce(
     (groupedUsers, user) => {
-      const status = presences[user.email] ? presences[user.email].aggregated.status : 'offline';
+      const status =
+        presences[user.email] &&
+        new Date().getTime() / 1000 -
```
Comment (borisyankov) [inline src/users/userHelpers.js]: I am sure you can simplify this and make it more clear.

File: src/common/UserStatusIndicator.js [hunk 2]
```diff
@@ -35,7 +37,14 @@ export default class UserStatusIndicator extends PureComponent<Props> {
   render() {
     const { presence, style } = this.props;
 
-    if (!presence || !presence.aggregated) return null;
+    if (
+      !presence ||
+      !presence.aggregated ||
+      new Date().getTime() / 1000 - (presence.aggregated || NULL_PRESENCE_AGGREGATED).timestamp >
```
Comment (borisyankov) [inline src/common/UserStatusIndicator.js]: Use `Date.now()` instead.

File: src/baseSelectors.js [hunk 3]
```diff
@@ -2,10 +2,26 @@
...
+import { getAllMessages, getPresence } from './directSelectors';
 import { NULL_ARRAY } from './nullObjects';
+import config from './config';
 
 export const getPrivateMessages = createSelector(
   getAllMessages,
   messages => messages[allPrivateNarrowStr] || NULL_ARRAY,
 );
+
+export const getValidPresence = createSelector(getPresence, presences => {
```
Comment (borisyankov) [inline src/baseSelectors.js]: `baseSelectors` is intended for the very simplest getters.
This should be somewhere else.

Also I am not a fan of the name - not descriptive enough.
Having Date.now() in the selector is also problematic, as it makes it non-pure.
We should discuss that further.

**Comment by jainkuniya** [comment]
Thanks @zulipbot. 
Will update it tonight.

**Comment by borisyankov** [comment]
Trying to cleanup open PRs. 
Feel free to open this again later.

**Pull request closed without merging**